### PR TITLE
Data ingestion implementation

### PR DESCRIPTION
**Pull request by DimuthuMadushan**
Data ingestion implementation to ingest csv file provided data

### CODE REVIEW

File: d [hunk 1]
```diff
@@ -0,0 +1,89 @@
+[1mdiff --git a/src/main/java/io/datasearch/diseasedata/store/RequestHandler.java b/src/main/java/io/datasearch/diseasedata/store/RequestHandler.java[m
+[1mindex c21d1bd..e85e76c 100644[m
```
Comment (malinthar) [inline d:2]: Remove this file from the commit

File: src/main/java/io/datasearch/diseasedata/store/RequestHandler.java [hunk 2]
```diff
@@ -7,7 +7,7 @@
 import org.slf4j.LoggerFactory;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
-
+import io.datasearch.denguestore.ingestion.DataIngester;
```
Comment (malinthar) [inline src/main/java/io/datasearch/diseasedata/store/RequestHandler.java]: double-check the package name

File: src/main/java/io/datasearch/diseasedata/store/dengdipipeline/DengDIPipeLine.java [hunk 3]
```diff
@@ -7,7 +7,7 @@
 import org.geotools.data.DataStore;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-
+import io.datasearch.denguestore.ingestion.DataIngester;
```
Comment (malinthar) [inline src/main/java/io/datasearch/diseasedata/store/dengdipipeline/DengDIPipeLine.java]: Same here

Comment (malinthar) [inline src/main/java/io/datasearch/diseasedata/store/dengdipipeline/DengDIPipeLine.java]: pakage name should be io.datasearch.disesedata.store.*

File: src/main/java/io/datasearch/diseasedata/store/dengdipipeline/ingestion/data/FeatureData.java [hunk 4]
```diff
@@ -0,0 +1,189 @@
+package io.datasearch.denguestore.data;
```
Comment (malinthar) [inline src/main/java/io/datasearch/diseasedata/store/dengdipipeline/ingestion/data/FeatureData.java]: Why do we need the FeatureData class

File: src/main/java/io/datasearch/diseasedata/store/dengdipipeline/ingestion/data/FeatureData.java [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+    //             featureAttributes.append(attributes.get(i).get("attribute"));
+    //             if (i != attributes.size() - 1) {
+    //                 featureAttributes.append(",");
+    //             }
+    //         }
+    //         sft = SimpleFeatureTypes.createType(getTypeName(), featureAttributes.toString());
+    //     }
+    // }
+
+    public void buildQueries(List<Map<String, String>> queriesList) {
```
Comment (malinthar) [inline src/main/java/io/datasearch/diseasedata/store/dengdipipeline/ingestion/data/FeatureData.java]: This method is not required?

File: src/main/java/io/datasearch/diseasedata/store/dengdipipeline/ingestion/data/FeatureData.java [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+        return features;
+    }
+
+    @Override
+    public List<Query> getTestQueries() {
+        return queries;
+    }
+
+    @Override
+    public Filter getSubsetFilter() {
```
Comment (malinthar) [inline src/main/java/io/datasearch/diseasedata/store/dengdipipeline/ingestion/data/FeatureData.java]: This is not required for the purpose of this class right?

File: src/main/java/io/datasearch/diseasedata/store/dengdipipeline/ingestion/data/DiseaseData.java [hunk 7]
```diff
@@ -0,0 +1,63 @@
+package io.datasearch.denguestore.data;
```
Comment (malinthar) [inline src/main/java/io/datasearch/diseasedata/store/dengdipipeline/ingestion/data/DiseaseData.java]: package name shoud be compying with the file path
it should be like` io.datasearch.diseasedata.store.dengdipipeline.ingestion.data`
We can remove the` data` part as well. Look into this

File: src/main/java/io/datasearch/diseasedata/store/dengdipipeline/ingestion/data/DiseaseData.java [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+     * @param x1 bounding box max x value
+     * @param y1 bounding box max y value
+     * @param dateField date attribute name
+     * @param t0 minimum time, exclusive, in the format "yyyy-MM-dd'T'HH:mm:ss.SSSZ"
+     * @param t1 maximum time, exclusive, in the format "yyyy-MM-dd'T'HH:mm:ss.SSSZ"
+     * @param attributesQuery any additional query string, or null
+     * @return filter object
+     * @throws CQLException if invalid CQL
+     */
+    static Filter createFilter(String geomField, double x0, double y0, double x1, double y1,
```
Comment (malinthar) [inline src/main/java/io/datasearch/diseasedata/store/dengdipipeline/ingestion/data/DiseaseData.java]: We dont't need this

File: src/main/java/io/datasearch/diseasedata/store/dengdipipeline/ingestion/data/DiseaseData.java [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+/**
+ * Interface for Disease Data Features.
+ */
+public interface DiseaseData {
+
+    String getTypeName();
+    SimpleFeatureType getSimpleFeatureType();
+    List<SimpleFeature> getFeatureData();
+    List<Query> getTestQueries();
+    Filter getSubsetFilter();
```
Comment (malinthar) [inline src/main/java/io/datasearch/diseasedata/store/dengdipipeline/ingestion/data/DiseaseData.java]: Don't need this. We only need to execute inserting operations. We are not going to use this model for querying

File: src/main/java/io/datasearch/diseasedata/store/dengdipipeline/ingestion/data/DiseaseData.java [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+
+/**
+ * Interface for Disease Data Features.
+ */
+public interface DiseaseData {
+
+    String getTypeName();
+    SimpleFeatureType getSimpleFeatureType();
+    List<SimpleFeature> getFeatureData();
+    List<Query> getTestQueries();
```
Comment (malinthar) [inline src/main/java/io/datasearch/diseasedata/store/dengdipipeline/ingestion/data/DiseaseData.java]: Don't need

**Review by malinthar — CHANGES_REQUESTED** [review]
Fix checkstyle issues and commented ones :)

**Review by chanukaDissanayaka — APPROVED** [review]

**Review by chanukaDissanayaka — APPROVED** [review]

**Pull request merged**